apps/aehttp/test/aehttp_ws_test_utils.erl

```diff
@@ -0,0 +1,165 @@
+-module(aehttp_ws_test_utils).
+-behaviour(websocket_client).
+
+%% API
+-export([start_link/2,
+         send/4,
+         register_test_for_event/3,
+         unregister_test_for_event/3,
+         wait_for_event/2,
+         wait_for_event/3,
+         stop/1]).
+
+%% behaviour exports
+-export([
+         init/1,
+         onconnect/2,
+         ondisconnect/2,
+         websocket_handle/3,
+         websocket_info/3,
+         websocket_terminate/3
+        ]).
+
+-define(DEFAULT_EVENT_TIMEOUT, 5000).
+-define(DEFAULT_SUB_TIMEOUT, 1000).
+
+start_link(Host, Port) ->
+    WsAddress = "ws://" ++ Host ++ ":" ++ integer_to_list(Port) ++ "/websocket",
+    ct:log("connecting to ~p", [WsAddress]),
+    {ok, Pid} = websocket_client:start_link(WsAddress, ?MODULE, self()),
+    ok = wait_for_event(websocket, connected),
+    {ok, Pid}.
+    
+
+stop(ConnPid) ->
+    ok = websocket_client:stop(ConnPid).
+
+send(ConnPid, Target, Action, Payload) ->
+    Msg0 = #{target => Target,
+             action => Action},
+    Msg =
+        case Payload == [] orelse Payload == #{} of
+            true -> Msg0;
+            false -> maps:put(payload, Payload, Msg0)
+        end,
+    ct:log("Sending to server ~p", [Msg]),
+    websocket_client:cast(ConnPid, {text, jsx:encode(Msg)}),
+    ok.
+
+%% when an WS event occours, the registered process will receive a message in
+%% the mailbox. Message structure will be:
+%% {websocket_event, Origin::atom(), Action::atom()}
+%% {websocket_event, Origin::atom(), Action::atom(), Payload::map()}
+%% where the Origin and Action are the same as the ones for which the process
+%% had registered. This message is consumed by wait_for_event/2,3
```

Comment:
This means a test process cannot start two WS clients and discriminate the received messages by source WS connection. Though not needed for current tests so ok.